### PR TITLE
Support clang in the Test compiler and JitBuilder

### PR DESCRIPTION
**Pull request by noryb009**
This adds support for building the Test compiler and JitBuilder using clang/LLVM on Linux x86-64.

Issue: #465

### CODE REVIEW

File: fvtest/compilertest/build/toolcfg/gnu/common.mk [hunk 1]
```diff
@@ -26,18 +26,27 @@ SED_PATH?=sed
 AR_PATH?=ar
 PERL_PATH?=perl
 
+ifeq ($(OS),osx)
```
Comment (mstoodle) [inline fvtest/compilertest/build/toolcfg/gnu/common.mk]: This seems unintuitive enough that it deserves at least a comment to explain. "as" works for me on OSX. Is there some version of OSX/clang where "clang" is the right way to access the assembler (or have I misunderstood) ?

File: jitbuilder/build/toolcfg/gnu/common.mk [hunk 2]
```diff
@@ -21,18 +21,27 @@ SED_PATH?=sed
 AR_PATH?=ar
 PERL_PATH?=perl
 
+ifeq ($(OS),osx)
```
Comment (mstoodle) [inline jitbuilder/build/toolcfg/gnu/common.mk]: same comment here

**Comment by noryb009** [comment]
Despite `__ --version` being the same for `as` and `clang`, `as` doesn't appear to support the `-D` flag (at least, on the machine I have access to). I did add a comment to explain the code.

**Comment by mstoodle** [comment]
Thanks @noryb009 !

**Pull request merged**